gsl/span

```diff
@@ -386,6 +387,11 @@ public:
     {
     }
 
+    template<class ArrayElementType = std::add_pointer<element_type>>
+    constexpr span(const std::unique_ptr<ArrayElementType>& ptr, index_type count) : storage_(ptr.get(), count) {}
```

Comment:
My feeling is that we want two constructors here, to distinguish between the array and non-array holding versions of `unique_ptr'. The non-array version doesn't need a count, it can just be a convenient way to initialize a single-element `span` from that particular standard container.